records/2024-02-15.private-eks-cluster.md

```diff
@@ -0,0 +1,94 @@
+
+This also means that the API is exposed to malicious actors. Although authentication is required to actually access the system, this still leaves it vulnerable to zero day attacks, DDoS attacks, and other security threats. 
+
+This may not conform to AC-17 of ITSG-33 government security requirements.
+
+##### Integration With GitHub
+
+Being a public endpoint, it is possible to provide GitHub direct connectivity to the cluster API. This has the benefit of simplifying release management with our CICD workflows in GitHub Actions. By granting GitHub access, any Kubernetes or Helm changes can be executed from the public GitHub Action Runners.
+
+##### Administrative Overhead
+
+There is little to no administrative overhead when using a public endpoint since this is the default state for AWS EKS clusters. 
+
+### Private EKS Cluster
+
+Using a private EKS Cluster requires the use of a VPN to connect to the cluster for administrative tasks. When the cluster is deployed, it will be unreachable unless connected via our AWS VPN endpoint for that particular environment.
+
+
+```mermaid
+graph TD;
+    EKS-->Internet-->GitHub;
+    Administrator-->VPN-->EKS;
+```
+   
+#### Considerations
+
+##### Access Control
+
+Access control is provided by using the CDS AWS SSO solution. A user that has authenticated with AWS and has the necessary permissions is able to log in and administer the cluster.
+
+##### Security
+
+Because the API endpoint is no longer exposed to the internet, any user needing to administer the cluster would have to use the appropriate GC Notify VPN connection. This would limit the ability of providing support to devices that have already been set up to connect to this VPN system. 
+
+This also means that the API will be shielded from any external threats, and in a better position to mitigate zero day attacks. 
+
+This is typically a requirement to satisfy AC-17 of ITSG-33 government security requirements.
+
+##### Integration With GitHub
+
+Running GitHub Actions that need to perform Kubernetes related tasks is more complex due to no direct connectivity to the EKS Cluster. Instead, in order to leverage GitHub Actions, it is required to install the [GitHub Actions Runner Controller](https://github.com/actions/actions-runner-controller) (ARC) into the target Kubernetes cluster. This turns the CICD model from push to pull in that ARC will poll the necessary repositories and when it detects changes, will create a GitHub Actions Runner **inside** the Kubernetes cluster. This has been completed as a proof of concept in the GC Notify Development Environment. 
+
+At a high level, the following steps are required:
+1. Generate a Personal Access Token in GitHub with the necessary permissions
+1. Install the GitHub Actions Runner Controller as a deployment in the Cluster (Helm)
```

Comment:
I don't understand why *Helm* is in parenthesis, we mean to do it via Helm? and for the next line too.